herana/models.py

```diff
@@ -50,16 +53,45 @@ def __unicode__(self):
         return self.statement
 
 
-class Faculty(models.Model):
+class OrgLevel(models.Model):
     name = models.CharField(max_length=256)
     institute = models.ForeignKey('Institute')
 
+    class Meta:
+        verbose_name = _('Org Level')
+        verbose_name_plural = _("Org Levels")
+        ordering = ['name']
+
+
+class OrgLevel1(OrgLevel):
     def __unicode__(self):
-        return self.name
+        return '%s - %s' % (self.institute.name, self.name)
 
     class Meta:
-        verbose_name_plural = _("Faculties")
-        ordering = ['name']
+        verbose_name = _('Org Level 1')
+        verbose_name_plural = _('Org Level 1')
+
+
+class OrgLevel2(OrgLevel):
+    parent = models.ForeignKey('OrgLevel1', verbose_name='Org Level 1')
+
+    def __unicode__(self):
+        return '%s - %s - %s' % (self.institute.name, self.parent.name, self.name)
+
+    class Meta:
+        verbose_name = _('Org Level 2')
+        verbose_name_plural = _('Org Level 2')
+
+
+class OrgLevel3(OrgLevel):
+    parent = models.ForeignKey('OrgLevel2', verbose_name='Org Level 2')
+
+    def __unicode__(self):
+        return '%s - %s - %s' % (self.institute.name, self.parent.name, self.name)
```

Comment:
This isn't going to show the org level 1 name. Perhaps you could do this recursively for org levels 2 and 3:

```python
return '%s - %s' % (unicode(self.parent), self.name)
```